src/main/java/edu/kit/compiler/transform/FirmGraphFactory.java

```diff
@@ -0,0 +1,53 @@
+package edu.kit.compiler.transform;
+
+import edu.kit.compiler.data.ast_nodes.ProgramNode;
+import edu.kit.compiler.lexer.StringTable;
+import edu.kit.compiler.semantic.NamespaceMapper;
+import firm.Construction;
+import firm.Dump;
+import firm.Entity;
+import firm.Graph;
+import firm.nodes.Node;
+import lombok.Data;
+
+import java.io.IOException;
+import java.util.UUID;
+
+@Data
+public class FirmGraphFactory {
+
+    private final NamespaceMapper namespaceMapper;
+    private StringTable stringTable;
+    private Graph graph;
+
+    public FirmGraphFactory(NamespaceMapper namespaceMapper, StringTable stringTable) {
+        this.namespaceMapper = namespaceMapper;
+        this.stringTable = stringTable;
+    }
+
+    public Graph createGraphFromAst(ProgramNode ast) {
+        TypeMapper entityMapper = new TypeMapper(namespaceMapper, stringTable);
```

Comment:
You should not be constructing a new `TypeMapper` here. It must only be created once, as it changes global state in Firm.